pandas/core/groupby/generic.py

```diff
@@ -1119,7 +1120,7 @@ def _aggregate_frame(self, func, *args, **kwargs) -> DataFrame:
         axis = self.axis
         obj = self._obj_with_exclusions
 
-        result: OrderedDict = OrderedDict()
+        result: Dict[Union[int, str], Union[NDFrame, np.ndarray]] = {}
```

Comment:
How is the key here a `str`? I think just int to hold position no?

Reply:
Seems like there are cases where it can also be a string. (`test_preserve_categories` in `test_categorical.py`)

```
categories = list("abc")
df = DataFrame({"A": Categorical(list("ba"), categories=categories, ordered=True)})
df.groupby("A", sort=True, observed=False)
```